InternetHospital.BusinessLogic/Services/DoctorService.cs

```diff
@@ -248,5 +253,130 @@ private bool FillIllness(IllnessHistoryModel illnessModel, Appointment appointme
             return result;
         }
 
+        /// <summary>
+        /// Get all patients who had appointments for current doctor
+        /// </summary>
+        /// <param name="doctorId"></param>
+        /// <returns>
+        /// returns a collection of doctor's patients
+        /// </returns>
+        private IQueryable<User> GetDateOfMyPatients(int doctorId)
+        {
+            var patients = _context.Users.Where(x => x.Appointments.Any(a => a.DoctorId == doctorId))
+                .Select(p => new User
+                {
+                    Id = p.Id,
+                    FirstName = p.FirstName,
+                    SecondName = p.SecondName,
+                    ThirdName = p.ThirdName,
+                    Email = p.Email
+                });
+            return patients;
+        }
+
+        /// <summary>
+        /// Convert patients collection from User model to MyPatientModel
+        /// </summary>
+        /// <param name="patients"></param>
+        /// <returns>
+        /// return collection of patients
+        /// </returns>
+        private IQueryable<MyPatientModel> ConvertToViewModel(IQueryable<User> patients)
+        {
+            var _patients = patients.Select(p => new MyPatientModel
```

Comment:
You can use automapper and remove this method.

Reply:
I deleted the method and moved the code to this method. AutoMapper does not work with IQueryable